src/entities/OrderDetailTax.java

```diff
@@ -0,0 +1,34 @@
+package entities;
+
+public class OrderDetailTax extends BaseEntity<Long>{
```

Comment:
You are missing Tax entity here

Reply:
Should i also add the original Tax entity that it represents as an attribute? 
For example, like in OrderDetailProduct, theres a product attribute of the original product that it represents

Reply:
Yes you should be able to compare current tax with prev tax

Reply:
solved